### ISSUE TITLE
SNIProxy fails to start

### ISSUE THREAD

**Issue by RavenX8** [lancachenet/sniproxy#6]
So I finially got around setting up a cache server for my network. Everything is working fine except for the sniproxy. When it starts up it crashes with the following error:

```
9/19/2018 2:47:35 AM2018-09-19 06:47:35 socket failed: Address family not supported by protocol
9/19/2018 2:47:35 AM2018-09-19 06:47:35 Failed to initialize listener [::]:443
```
I'm not sure what to do here as most of my searching google comes up empty. Any ideas on how to fix this?

Docker 18.06.1-ce
Ubuntu 17.10 (4.13.0)

**Comment by JasonRivers** [comment]
Please could you show the docker run command you used?

**Comment by RavenX8** [comment]
```
root@lancache:/lancache# docker run --name sniproxy -p 443:443 steamcache/sniproxy:latest
2018-09-19 12:17:41 socket failed: Address family not supported by protocol
2018-09-19 12:17:41 Failed to initialize listener [::]:443
root@lancache:/lancache# docker logs sniproxy
2018-09-19 12:17:41 socket failed: Address family not supported by protocol
2018-09-19 12:17:41 Failed to initialize listener [::]:443
root@lancache:/lancache# docker top sniproxy
Error response from daemon: Container cc3711e969d942822aef66e6a0ee263e14692295d677738e753ee7a452ded9af is not running
root@lancache:/lancache# netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 0.0.0.0:sunrpc          0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:ssh             0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:hostmon         0.0.0.0:*               LISTEN
udp        0      0 0.0.0.0:hostmon         0.0.0.0:*
udp        0      0 127.0.0.53:domain       0.0.0.0:*
udp        0      0 192.168.1.69:bootpc     0.0.0.0:*
udp        0      0 0.0.0.0:sunrpc          0.0.0.0:*
udp        0      0 0.0.0.0:845             0.0.0.0:*
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node   Path
unix  2      [ ACC ]     SEQPACKET  LISTENING     12662    /run/udev/control
unix  2      [ ACC ]     STREAM     LISTENING     755093   /run/user/1000/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     26429    @/containerd-shim/moby/4f9fc81775bd3306b2c170f915f6d443bf079dc78521bc5c88b3e6c72286840e/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     19353    /var/run/docker/containerd/docker-containerd-debug.sock
unix  2      [ ACC ]     STREAM     LISTENING     19354    /var/run/docker/containerd/docker-containerd.sock
unix  2      [ ACC ]     STREAM     LISTENING     20773    /var/run/docker/metrics.sock
unix  2      [ ACC ]     STREAM     LISTENING     29141    @/containerd-shim/moby/f31aae128659767ba5e4a1d34e2c8de687ae0df0aa70273d21af06350b97dc32/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     25386    @/containerd-shim/moby/2d6f7278aa55d3d0c49ecfc6c4224e96ecd3b0ab771985cef8907ce15425e8fc/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     26653    @/containerd-shim/moby/11cff743890ad9d9e79fd0139e67bc619a3e67ae43c98cf2b9059e9c9e54a436/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     12655    /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     12661    /run/rpcbind.sock
unix  2      [ ACC ]     STREAM     LISTENING     34719    @/containerd-shim/moby/406243d0c9835cb307f0b04553905e3ec76b2c12bb2fa90c6cc1140dc9441970/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     12672    /run/systemd/fsck.progress
unix  2      [ ACC ]     STREAM     LISTENING     26370    @/containerd-shim/moby/56532e9fbea2cc31d4a2e016097f9c0b4f55b0abf039ca462cdf9fc572f97b85/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     169196   @/containerd-shim/moby/992327c655ce28909577e7cf7bdf18e42110131c15bdd6c1e429121cc793d613/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     12678    /run/lvm/lvmpolld.socket
unix  2      [ ACC ]     STREAM     LISTENING     38000    @/containerd-shim/moby/fbcf6ac3b1dc5e5c57421f198c47ddcb2280100f9eaf0a0fe7e2df9a5ec19731/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     29098    @/containerd-shim/moby/85f31e0fa08b1088e23de8b7815d0d371e1baf98aee4d5a5267efe33aca7c886/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     12679    /run/systemd/journal/stdout
unix  2      [ ACC ]     STREAM     LISTENING     12686    /run/lvm/lvmetad.socket
unix  2      [ ACC ]     STREAM     LISTENING     22815    /run/docker/libnetwork/36704e4bf8ced297f40a6a13a70a7206b4e60f66ea11b26624775f30eed618ac.sock
unix  2      [ ACC ]     STREAM     LISTENING     23118    /tmp/log.sock
unix  2      [ ACC ]     STREAM     LISTENING     30864    @/containerd-shim/moby/e79c43a9d37d7626220eb934cc932dddd8c7eb705fdf0826c22f3daaaa32ed1b/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     38232    @/containerd-shim/moby/364f5a5e52182c12b007b622a7d13e9dc74d9fc1f6cb346eddee6b0405913c46/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     27850    @/containerd-shim/moby/9c973753b54c3b79475bd3ff2a98ee7a8234a09b6ef8889d8d8c0e2c0a40047c/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     30045    @/containerd-shim/moby/f171e163299e358722fd15f4627481959f8320d08ea53e895fa9e72db15c035d/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     25017    @/containerd-shim/moby/e57753ce53360fe656c6338a85f88272c0c7c34e57322900045d085e321f7b0e/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     36949    @/containerd-shim/moby/b82286f30510d90df2c8f5df12c0f35671c30d15af62c9cd9904b5b8a447e2f6/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     23447    @/containerd-shim/moby/5394a9da8c2e741502d8fb94ef167d9db3e7c286d86b72216f7cb215b5c4568c/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     13077    /var/lib/lxd/unix.socket
unix  2      [ ACC ]     STREAM     LISTENING     23644    @/containerd-shim/moby/120f6cc87576dfc1f847b768b17e0f778095b0e9e3639afb0c1e8e6a4a3302d2/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     29753    @/containerd-shim/moby/8bf88a20836ddb13f78932035af7c1fbf833625096f1eeb6705d4d6ce9c32b65/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     50697    @/containerd-shim/moby/36838e831eee39c24031f4df0e38e1decd27ffb5f3d094305e363d4c94076c16/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     27562    @/containerd-shim/moby/ad5f73215db96aa849dd47fd06260c8896bce03d9809c8faaf35a519040f3261/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     28100    @/containerd-shim/moby/ea11787ed54206d1e41905c57feb9ff8c7d4dd252ea66baf33407a9afcf6e47c/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     23429    /tmp/log.sock
unix  2      [ ACC ]     STREAM     LISTENING     37515    @/containerd-shim/moby/df2e3fdcec3ca1af0a3a44881f90922f680155002b214b3feb93fdcae22e9ca0/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     27963    @/containerd-shim/moby/19a1651af45ab32f975d30a20b2af68dc7115412f37e066a17847d7da8512488/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     24782    @/containerd-shim/moby/482ec65039368714e6e5fdc7f407f4f8d7f941eb7e8ae8900398fdd730cdde84/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     36673    @/containerd-shim/moby/2ea79ceff6b55046116c8275799a412d1192962e9297a35ab90e4f1dd4bb56a5/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     29975    @/containerd-shim/moby/f179f39136d910750efee71d23213e22754cf5ffabdbc43adf49cdb4876793ae/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     34684    @/containerd-shim/moby/a6d3cd61e0d5c6e0197c5203309b7fa431dbe44febc1bc95e3be6e8b370c9e37/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     25108    @/containerd-shim/moby/50f2b344f5f1231817d77310e399c3e80a97f9df49e9906b5f87dd41e92c3124/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     24919    @/containerd-shim/moby/3c59fc7e08b7a499ea0233d9d24f2fa101d37711fbb960a8a25bfb815c57c63b/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     159549   @/containerd-shim/moby/6426f33a2bead0eedac61dbcc2ca580528131fdab839c98c020d209c5a9bc459/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     36212    @/containerd-shim/moby/952c10cb9c224e9b92b68baa919bfddf8a4ad74047316df7485a6ec8d1c849cc/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     16879    @ISCSIADM_ABSTRACT_NAMESPACE
unix  2      [ ACC ]     STREAM     LISTENING     35753    @/containerd-shim/moby/9c96106466ac177d3ad9832bb95d25353c8c10c5d0f586981c9b045e6c26d9aa/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     24858    @/containerd-shim/moby/446ab4742e4edc7d3a83ba7083841bf374ac3fa52ebda7fa92aedbf349eb59ff/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     25226    @/containerd-shim/moby/df1d3c0f81748515679f927026d7e7cab97c8e43e4afabb2c904f6f317aef355/shim.sock@
unix  2      [ ACC ]     STREAM     LISTENING     13071    /run/uuidd/request
unix  2      [ ACC ]     STREAM     LISTENING     13072    /run/acpid.socket
unix  2      [ ACC ]     STREAM     LISTENING     13073    /run/snapd.socket
unix  2      [ ACC ]     STREAM     LISTENING     13074    /run/snapd-snap.socket
unix  2      [ ACC ]     STREAM     LISTENING     13075    /var/run/dbus/system_bus_socket
unix  2      [ ACC ]     STREAM     LISTENING     13076    /var/run/docker.sock

```

**Comment by JasonRivers** [comment]
Thanks.
This seems to be because you have IPv6 disabled in your kernel.
Is there a reason that you have IPv6 disabled in the kernel?

We will look in to this.

**Comment by RavenX8** [comment]
it's disabled because everything (mostly apt) will default to IPv6 which is an issue for me since when using IPv6, it will take about 3 minutes to resolve a each repo. I wanted to update to ubuntu 18, because it fixes the issue, but when I boot, it drops into initramfs which is a known bug for freenas. 

I'll try enabling it again.

**Comment by JasonRivers** [comment]
Sounds like a broken IPv6 route on your network.
In the meantime, could you test the image I just pushed with the following:
```
docker run --name sniproxy -p 443:443 steamcache/sniproxy:disablev6
```
If this works please could you report back here.

**Comment by JasonRivers** [comment]
This looks to be introduced in commit a0eed8f7 from @kixelated the cleanup was perhaps a little too much if we want to force IPv4.

**Comment by RavenX8** [comment]
I have tested your new image and can confirm that it now works no issues

**Comment by JasonRivers** [comment]
This is now resolved as of the merge for PR #7 you should be able to pull latest image again once docker hub have built it.
